modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -3072,11 +3072,22 @@ else if (sqlType == Types.DECIMAL) {
 				value = BigDecimal.ZERO;
 			}
 
+			String decimalValue = String.valueOf(value);
```

Comment:
I believe you can try to take advantage of some useful tools, like **DecimalFormat** and **NumberUtil** to perform that. See [here](https://github.com/liferay/liferay-portal/blob/213dd7c93ca88db1cc68455c17001e81ed70afa0/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/util/NumericDDMFormFieldUtil.java#L29-L62)